libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -128,33 +127,40 @@ public static synchronized SyncManager getInstance(UserAccount account) {
      * @return Instance of this class.
      */
     public static synchronized SyncManager getInstance(UserAccount account, String communityId) {
+        return getInstance(account, communityId, null);
+    }
+
+    /**
+     * Returns the instance of this class associated with this user, community and smartstore.
+     *
+     * @param account User account. Pass null to user current user.
+     * @param communityId Community ID. Pass null if not applicable
+     * @param smartStore SmartStore instance. Pass null to use current user default smartstore.
+     *
+     * @return Instance of this class.
+     */
+    public static synchronized SyncManager getInstance(UserAccount account, String communityId, SmartStore smartStore) {
         if (account == null) {
             account = SalesforceSDKManagerWithSmartStore.getInstance().getUserAccountManager().getCurrentUser();
         }
 
-        SmartStore smartStore = SmartSyncSDKManager.getInstance().getSmartStore(account, communityId);
-        SyncManager syncManager = getInstanceForStore(smartStore);
+        if (smartStore == null) {
+            smartStore = SmartSyncSDKManager.getInstance().getSmartStore(account, communityId);
+        }
 
-        // Setting up rest client
-        RestClient restClient = SalesforceSDKManager.getInstance().getClientManager().peekRestClient(account);
-        syncManager.setRestClient(restClient);
+        if (UserAccount.INTERNAL_COMMUNITY_ID.equals(communityId)) {
+            communityId = null;
+        }
+        String uniqueId = (account != null ? account.getUserId() : "") + ":"
```

Comment:
We can take ```communityId``` out of this ```uniqueId``` computation, and with it, the line above that sets ```communityId = null```. The DB path accounts for ```userId, orgId, communityId```, and all we need to associate it with a user account is ```userId```. If a portal user if logged into a community, the ```userId``` will be different. If an internal user is logged into both org and community, the DB path will be different. Hence, the ```uniqueId``` should always be unique with ```userId + dbPath```.